src/firmware/boards/legacy_robot_stm32f4/io/receive.c

```diff
@@ -180,16 +179,9 @@ void receive_tick(log_record_t *record)
     // Decrement timeout tick counter if nonzero.
     if (timeout_ticks == 1)
     {
-        timeout_ticks = 0;
-        charger_enable(false);
-        chicker_discharge(true);
-
-        TbotsProto_Primitive primitive_msg = TbotsProto_Primitive_init_zero;
-        primitive_msg.which_primitive      = TbotsProto_Primitive_stop_tag;
-
-        xSemaphoreTake(drive_mtx, portMAX_DELAY);
-        app_primitive_manager_startNewPrimitive(primitive_manager, world, primitive_msg);
-        xSemaphoreGive(drive_mtx);
+        timeout_ticks = 1;
+        // Make robot safe by discharging through chicker and stopping all motors
+        app_primitive_makeRobotSafe(world);
```

Comment:
So I know here we are effectively "triggering estop behaviour" when we timeout. Is it easy enough to just start/run an estop primitive using the primitive manager here? That way if the estop implementation changes it gets updated in all places. Then we don't risk different behaviour when the robot times out vs a manually triggered estop.

Reply:
I think that it's best not to involve the primitive manager if possible because that's a lot of complexity to handle this case